src/Hazelcast.Net/Clustering/ClusterConnections.cs

```diff
@@ -547,25 +543,30 @@ private async Task ConnectFirstAsync(CancellationToken cancellationToken)
                 {
                     canRetry = false; // retry strategy was canceled
                 }
-                catch (Exception e) // gather exceptions
+                catch (Exception e)
                 {
-                    exceptions ??= new List<Exception>();
-                    exceptions.Add(e);
+                    _logger.LogError(e, "Connection attempt has thrown.");
                     canRetry = false; // retry strategy threw
+                    isExceptionThrown = true;
                 }
 
             } while (canRetry);
 
-            var aggregate = exceptions == null ? null : new AggregateException(exceptions);
+            string msgSomeThingWentWrong = "";
+
+            if (isExceptionThrown)
+                msgSomeThingWentWrong = "Some exceptions were thrown and have been written to the log. Please refer to the log for details.";
 
             // canceled exception?
             if (cancellationToken.IsCancellationRequested)
                 throw new OperationCanceledException($"The cluster connection operation to \"{_clusterState.ClusterName}\" has been canceled. " +
-                    $"The following addresses where tried: {string.Join(", ", tried)}.", aggregate);
+                    $"The following addresses where tried: {string.Join(", ", tried)}." +
+                    $" {msgSomeThingWentWrong}");
 
             // other exception
             throw new ConnectionException($"Unable to connect to the cluster \"{_clusterState.ClusterName}\". " +
-                $"The following addresses where tried: {string.Join(", ", tried)}.", aggregate);
```

Comment:
.NET uses `ConnectionException` when could not connect and Java uses `IllegalStateException`. This is confusing. Why is this difference? (it may not related to this PR)

Reply:
Isn't `ConnectionException`  more relevant to the context? WDYT @zpqrtbnk?

Reply:
Yes, .NET uses different names for a range of exceptions, so if you compare line-to-line to Java, you're going to be confused. The `ConnectionException` name was choosen because it was considered more meaningful than the Java couterpart (illegal state?).

This is not related to this PR, so should not block the PR. OTOH we need to have one discussion re. the differences between the .NET Client and Java Client. They *are* definitively different, the more you look into it the more you'll be confused. And we need to decide whether this is acceptable, or whether we want to start aligning things.

Reply:
Yes, it should not stop the PR. My take on the matter is that unless there is a good reason and very obvious benefit to the user, we try keeping things the same so that the users will be more familiar with names, etc. and better documentation for all the clients. Let's talk about this in the group.